autonat.go

```diff
@@ -61,12 +63,18 @@ type AmbientAutoNAT struct {
 }
 
 // NewAutoNAT creates a new ambient NAT autodiscovery instance attached to a host
-func NewAutoNAT(ctx context.Context, h host.Host) AutoNAT {
+func NewAutoNAT(ctx context.Context, h host.Host, ga ...GetAddrs) AutoNAT {
+	getAddrs := h.Addrs
+	if len(ga) > 0 {
```

Comment:
I'd just make this a static arg and check if it's nil

Reply:
ok

Reply:
done.

Reply:
Why not a variadic `Option`? @magik6k @vyzo